File: designer/src/main/java/com/skm/sftp/designer/DesignerHook.java
package com.skm.sftp.designer;

import com.inductiveautomation.ignition.common.script.ScriptManager;
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider;
import com.inductiveautomation.ignition.designer.model.AbstractDesignerModuleHook;
import com.skm.sftp.client.ClientScriptModule;

@SuppressWarnings("unused")
public class DesignerHook extends AbstractDesignerModuleHook {
  @Override
  public void initializeScriptManager(ScriptManager manager){
    super.initializeScriptManager(manager);

    manager.addScriptModule(
        "system.sftp",
        new ClientScriptModule(),
        new PropertiesFileDocProvider());
  }
}

File: designer/src/main/kotlin/com/skm/sftp/designer/DesignerHook.kt


package com.skm.sftp.designer

import com.inductiveautomation.ignition.common.script.ScriptManager
import com.inductiveautomation.ignition.common.script.hints.PropertiesFileDocProvider
import com.inductiveautomation.ignition.designer.model.AbstractDesignerModuleHook
import com.skm.sftp.client.ClientScriptModule
import com.skm.sftp.modulePath


@Suppress("unused")
class DesignerHook : AbstractDesignerModuleHook() {
    override fun initializeScriptManager(manager: ScriptManager) {
        super.initializeScriptManager(manager)
        manager.addScriptModule(
            modulePath,
            ClientScriptModule(),
            PropertiesFileDocProvider()
        )
    }
}